### PR TITLE
Upgrade patch level dependencies 

### PR DESCRIPTION
**Pull request by christophermanning**
- Upgrade go to 1.5.4
- Upgrade hub to 2.2.3
- Specify tag for go-makefile Docker image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 FROM ubuntu:14.04
 
-ENV GOVERSION 1.5.3
-
 RUN apt-get update -y && apt-get install --no-install-recommends -y -q \
                          curl build-essential ca-certificates git mercurial bzr \
                       && rm -rf /var/lib/apt/lists/*
 
+ENV GOVERSION 1.5.4
```
Comment (mattyb) [inline Dockerfile:7]: Could we set this just before it's needed, so that more of the layers will be reused?

Comment (christophermanning) [inline Dockerfile:7]: That's why I moved it below the long running apt-get commands; the only things between where it is now and where it's used are just ENV and directory commands. Are those worth optimizing the reuse of?

Comment (mattyb) [inline Dockerfile:7]: You're right, the layers are just diffs, so it shouldn't make a significant difference

**Comment by mattyb** [comment]
Is it expected that travis is failing?

**Comment by christophermanning** [comment]
Build is green.

**Comment by mattyb** [comment]
LGTM

**Pull request merged**